aqsconverters/aq.py

```diff
@@ -48,12 +55,122 @@ def produce_annotation(self, aq_query_type, *args, **kwargs):
+                                           + radius_obj_id_suffix,
+                                       name=radius_arg_str)
+                    astro_image_name += '_' + radius_obj.name
+
+            # pixels
+            pixels_obj = None
+            if 'pixels' in kwargs:
+                pixels = kwargs['pixels']
+                if pixels is not None:
+                    pixels_obj_id_suffix = hashlib.sha256(str(pixels).encode()).hexdigest()
+                    pixels_obj = Pixels(_id="https://odahub.io/ontology#Pixels/"
+                                           + pixels_obj_id_suffix,
+                                       name=str(pixels))
+                    astro_image_name += '_' + pixels_obj.name
+                    astro_image_suffix += '_' + str(pixels)
+
+            # image_band
+            image_band_obj = None
+            if 'image_band' in kwargs:
+                image_band = kwargs['image_band']
+                if image_band is not None:
+                    image_band_obj_id_suffix = hashlib.sha256(image_band.encode()).hexdigest()
+                    image_band_obj = Pixels(_id="https://odahub.io/ontology#ImageBand/"
+                                           + image_band_obj_id_suffix,
+                                       name=image_band)
+                    astro_image_name += '_' + image_band_obj.name
+                    astro_image_suffix += '_' + image_band
+
+            astro_image_suffix = hashlib.sha256(astro_image_suffix.encode()).hexdigest()
+
+            astro_image_obj = AstrophysicalImage(_id="https://odahub.io/ontology#AstroImage/"
+                                                       + astro_image_suffix,
+                                                 name=astro_image_name)
+
+            astro_image_obj.isUsingSkyCoordinates = [skycoord_obj]
+
+            if radius_obj is not None:
+                astro_image_obj.isUsingRadius = [radius_obj]
+
+            if image_band_obj is not None:
+                astro_image_obj.isUsingImageBand = [image_band_obj]
+
+            if pixels_obj is not None:
+                astro_image_obj.isUsingPixels = [pixels_obj]
+
+            run.isRequestingAstroImage = [astro_image_obj]
+
         if aq_query_type == "query_object":
             obj_name = args[0]
             obj =  AstrophysicalObject(_id="https://odahub.io/ontology#AstroObject" + obj_name.replace(" ","_"),
                                        name=obj_name) # normalize id
             run.isRequestingAstroObject = [obj]
 
+        if aq_query_type == "query_region":
+            if 'coordinates' in kwargs:
+                coordinates_arg = kwargs['coordinates']
+            else:
+                coordinates_arg = args[0]
+
+            skycoord_obj_id_suffix = hashlib.sha256(coordinates_arg.to_string().encode()).hexdigest()
+            skycoord_obj = SkyCoordinates(_id="https://odahub.io/ontology#SkyCoordinates/"
```

Comment:
please no "/" at the end, it's not conventional.

Reply:
The resulting `_id` will be like the following:

```
https://odahub.io/ontology#SkyCoordinates/fcc7c6ac1b62f84948cd4a19b2e4fa76033eec52f021777e32e8fc5e4a73c3e6
```

There is no '/' at the end. Or are you referring to the one after SkyCoordinates?

Reply:
yes, after the sign, in the fragment. It confuses with uri path which is before. It may cause inconsistent treatment.